src/ripple/basics/ScopedLock.h

```diff
@@ -61,40 +64,40 @@ namespace ripple
 
     }  // mut gets unlocked here
     @endcode
-
 */
 template <class MutexType>
-class GenericScopedUnlock
+class ScopedUnlock
 {
-    MutexType& lock_;
+    std::unique_lock<MutexType>& lock_;
 public:
-    /** Creates a GenericScopedUnlock.
+    /** Creates a ScopedUnlock.
 
-        As soon as it is created, this will unlock the CriticalSection, and
-        when the ScopedLock object is deleted, the CriticalSection will
+        As soon as it is created, this will unlock the unique_lock, and
+        when the ScopedLock object is deleted, the unique_lock will
         be re-locked.
 
         Make sure this object is created and deleted by the same thread,
         otherwise there are no guarantees what will happen! Best just to use it
-        as a local stack object, rather than creating one with the new() operator.
+        as a local stack object, rather than creating on the heap.
     */
-    explicit GenericScopedUnlock (MutexType& lock) noexcept
+    explicit ScopedUnlock (std::unique_lock<MutexType>& lock)
```

Comment:
In C++17 this constructor implicitly provides the class template argument deduction guide that you need elsewhere.